lib/rdf2marc/model2marc/control_field_008.rb

```diff
@@ -30,6 +30,29 @@ def value
 
       private
 
+      # @param [String] code from https://www.loc.gov/marc/geoareas/gacs_code.html
+      # @return [String] a three letter code from https://id.loc.gov/vocabulary/countries.html
+      def place_from_geo_area_code(code)
+        return 'xx ' unless code
+        return lookup_country_from_geoarea(code) if in_country_authority?(code)
+
+        code[0, 3]
+      end
+
+      def lookup_country_from_geoarea(code)
```

Comment:
Lookups / resolving should be done as part of rdf2model, not model2marc.

Reply:
This seems like an important design decision. Is this documented somewhere?

Reply:
In the code? (I'm open to other suggestions).

All resolving/lookup (such as this) is handled by a Resolver, in this case https://github.com/LD4P/rdf2marc/blob/main/lib/rdf2marc/resolver/id_loc_gov_resolver.rb.

That resolving is performed in rdf2model, for example, https://github.com/LD4P/rdf2marc/blob/main/lib/rdf2marc/rdf2model/mappers/number_and_code_fields.rb#L105.

Reply:
I was thinking of a design doc.